src/store/modules/position.store.js

```diff
@@ -312,15 +312,18 @@ const actions = {
      *   of the debug console)
      */
     setCameraPosition({ commit }, { position, dispatcher }) {
-        const wrappedPosition = {
-            x: position.x,
-            y: position.y,
-            z: position.z,
-            // wrapping all angle-based values so that they do not exceed a full-circle value
-            roll: wrapDegrees(position.roll),
-            pitch: wrapDegrees(position.pitch),
-            heading: wrapDegrees(position.heading),
-        }
+        const wrappedPosition =
+            position && Object.keys(position)?.length === 6
+                ? {
+                      x: position.x,
+                      y: position.y,
+                      z: position.z,
+                      // wrapping all angle-based values so that they do not exceed a full-circle value
+                      roll: wrapDegrees(position.roll),
+                      pitch: wrapDegrees(position.pitch),
+                      heading: wrapDegrees(position.heading),
+                  }
+                : null
```

Comment:
Why this ? I'm not sure that setting a null position is a very good idea ?

Reply:
null is the camera value everytime we are in 2d 
https://github.com/geoadmin/web-mapviewer/blob/7f7397ceef45213da51192bb710795321fe8309c/src/store/modules/position.store.js#L104-L111

I encountered a bug where, when switching from 3d to 2d, I got an error because there was no check to see if the camera received a valid input (it was trying to do `null.x` , which does not work.)

I saw that on `develop` there was an error too, but that error wasn't blocking. It was blocking locally on my branch, so I thought I could correct this. We can make this a separate bugfix if you'd prefer and add a comment in the function as well if you think it'd  be clearer as to why we are doing this.